### PR TITLE
feat: add exists() predicate to BackendProtocol

### PR DESCRIPTION
**Pull request by OchnikBartek**
## Summary

Adds a public `exists(path: str) -> bool` predicate to `BackendProtocol` so consumers can reliably check file presence without sniffing private attributes (e.g. `StateBackend._files`) or pattern-matching empty-byte returns from `_read_bytes()`. Every concrete backend gets a fresh implementation matching the protocol's "exists() = exists as a regular file" semantics; directories and invalid paths return False.

The contract gap this fixes: `StateBackend._read_bytes` silently returns `b""` for missing files — indistinguishable from a real empty file. Without `exists()`, consumers fall back to private dict access (`path in backend._files`) which breaks encapsulation. Test `test_exists_distinguishes_empty_file_from_missing` demonstrates this explicitly.

## Changes

- `protocol.py` — new `exists()` method, added as the first method on `BackendProtocol` (semantically precedes `read`)
- `backends/state.py` — dict-lookup after `_validate_path` / `_normalize_path` helpers
- `backends/local.py` — `Path.is_file()` after `_validate_path`; only `PermissionError` is reachable (documented in docstring)
- `backends/composite.py` — one-line delegate to `_get_backend(path).exists(path)`
- `backends/base.py` — shell-based default for every `BaseSandbox` subclass via `test -f` with `shlex.quote` (Docker inherits it automatically)
- `backends/daytona.py` — native override using `self._sandbox.fs.get_file_info()`; broad `except Exception` matches the file's existing pattern (`_read_bytes`, `write`, etc.) and is documented in the docstring; returns `False` on any failure or when `is_dir` is true

## Testing

- [x] New tests added for any new functionality (required — `make test` enforces 100% coverage)
- [x] All tests pass locally: `make test`
- [x] Linting passes: `make lint`
- [x] Type checking passes: `make typecheck`
- [x] Security scan passes: `make security`
- [x] Full check suite passes: `make all`

**Test coverage details:**
- `StateBackend`: 4 tests including the **empty-file-vs-missing distinction** (`_read_bytes` returns `b""` for both, `exists()` returns `True`/`False` respectively — this is the smoking gun for the motivation)
- `LocalBackend`: 4 tests including the directory edge case (`is_file()` correctly returns `False` for dirs)
- `CompositeBackend`: 3 tests including a cross-backend routing assertion (`composite.exists("/special/x")` is `True` while `default.exists("/special/x")` is `False` — proves routing actually happens)
- `DaytonaSandbox`: 4 tests via stubbed `FakeFs.get_file_info` + new `FakeFileInfo` dataclass
- `DockerSandbox`: 3 `@pytest.mark.docker`-gated tests covering existing file, missing file, and directory

Final stats: 477 passed, 18 deselected (Docker infra), 100% coverage maintained, pyright 0 errors, mypy clean.

## Related Issues
- Epic [#101 — Live Run Forking](https://github.com/vstorm-co/pydantic-deepagents/issues/101)
- Sub-issue [#102 — Stage 1: LiveForkCapability core](https://github.com/vstorm-co/pydantic-deepagents/issues/102)

### CODE REVIEW

File: tests/test_docker_sandbox.py [hunk 1]
```diff
@@ -411,11 +411,28 @@ def test_read_file_nonexistent_path(self, docker_sandbox):
         assert "not found" in content
 
     @pytest.mark.docker
-    def test_read_bytes_nonexistent_path(self, docker_sandbox):
-        """_read_bytes on a non-existent path returns empty bytes."""
-        result = docker_sandbox._read_bytes("/workspace/no_such_file.bin")
+    def testread_bytes_nonexistent_path(self, docker_sandbox):
```
Comment (DEENUU1) [inline tests/test_docker_sandbox.py]: The `_read_bytes` → `read_bytes rename` was applied with a boundary-less search/replace and accidentally rewrote test function names: `test_read_bytes` became `testread_bytes`, and `test_edit_read_bytes_returns_error_bytes` became 
`test_editread_bytes_returns_error_bytes`. These still get collected (they match pytest's test* glob), so the suite is
green by luck, but the names are now mangled and unreadable. 

Please redo the rename with word boundaries so it only touches the _read_bytes method/symbol, and restore the original test names (test_read_bytes, test_read_bytes_string, test_edit_read_bytes_returns_error_bytes, etc.).

File: src/pydantic_ai_backends/backends/local.py [hunk 2]
```diff
@@ -220,6 +220,21 @@ def _validate_path(self, path: str) -> Path:
...
+    def exists(self, path: str) -> bool:
+        """Check whether a file exists on the filesystem.
+
+        ``_validate_path`` raises ``PermissionError`` when the resolved path
+        escapes the allowed directory set; that is the only documented
+        failure mode for path resolution and maps to ``False`` per the
+        protocol contract ("invalid paths, directories, and permission
+        errors all return False").
+        """
+        try:
```
Comment (DEENUU1) [inline src/pydantic_ai_backends/backends/local.py:243]: The protocol docstring promises "Returns False for invalid paths". `Path.is_file()` swallows most `OSErrors`, but it still raises ValueError for paths with an embedded null byte (e.g. exists("foo\x00bar")), which escapes this method.

**Review by DEENUU1 — CHANGES_REQUESTED** [review]
